### PR TITLE
10 add preview page

### PR DESCRIPTION
**Pull request by poludnev**

### CODE REVIEW

File: src/utils/index.ts [hunk 1]
```diff
@@ -167,3 +167,11 @@ export const getStampedImagePreviewDataUrl = (
 
   return canvas.toDataURL();
 };
+
+export const validateBigInt = (data: string): Promise<boolean> =>
```
Comment (Elizavetta) [inline src/utils/index.ts:171]: possibly we can use a sync version of validation ?

**Review by Elizavetta — APPROVED** [review]

File: extension/popup/Popup.tsx [hunk 2]
```diff
@@ -70,26 +71,59 @@ const Popup: React.FC = () => {
       });
   };
 
+  useEffect(() => {
+    chrome.tabs.query({ active: true, lastFocusedWindow: true }, (tabs) => {
+      const url = new URL(tabs[0].url!);
+      const host = url.host;
+      const currentPath = url.pathname;
+      const urlTweetId = currentPath.split('/').at(-1) ?? '';
```
[the remark this answers is not in the document]
Comment (poludnev) [reply extension/popup/Popup.tsx]: tabs[0].url = 'https://twitter.com/twitter/status/1628800860103778304?current=1';
new URL(tabs[0].url).pathname = '/twitter/status/1628800860103778304';
thus currentPath.split('/').at(-1) ?? '' === 1628800860103778304, but not '1628800860103778304?current=1'. 
that new URL just right for cutting out queries. 

We can have tweet urls with query params, but they are useless: twitter.com ignores queries, the result of https://twitter.com/poludnev/status/1635717753104367625?current=1 is completely same as https://twitter.com/poludnev/status/1635717753104367625, and twitter docs say the "canonical" url is without query, so seems not to have problems:
https://developer.twitter.com/en/blog/community/2020/getting-to-the-canonical-url-for-a-tweet

File: extension/popup/Popup.tsx [hunk 3]
```diff
@@ -70,26 +71,59 @@ const Popup: React.FC = () => {
...
+
+      if (host !== 'twitter.com' || !currentPath.includes('status')) {
+        setTweetId(null);
```
[the remark this answers is not in the document]
Comment (poludnev) [reply extension/popup/Popup.tsx:85]: ok!

File: extension/popup/popup.css [hunk 4]
```diff
@@ -1,3 +1,5 @@
+@import url('https://fonts.googleapis.com/css2?family=Chivo+Mono:wght@400;700&family=Cutive+Mono&display=swap');
```
[the remark this answers is not in the document]
Comment (poludnev) [reply extension/popup/popup.css]: fixed

File: src/components/MetadataPreview/MetadataPreview.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+import cn from 'classnames';
+import { IMetadataPreviewProps } from './MetadataPreviewProps';
+import classes from './MetadataPreview.module.scss';
+
+export const MetadataPreview: React.FC<IMetadataPreviewProps> = memo(
+  ({ data, preview, dnsHash, headersHash }) => {
+    const { ip, url, headers, dns } = data;
+    const headersKeys = Object.keys(headers);
+    const dnsData = dns.data
+      .filter((el: string) => el.length > 0 && !el.includes(';'))
```
[the remark this answers is not in the document]
Comment (poludnev) [reply src/components/MetadataPreview/MetadataPreview.tsx:12]: Add a comment with source of sample:  // sample data source: src/__fixtures__/metadata.ts
We exclude ";" just for removing the first two strings of dns info:
'; <<>> DiG 9.10.6 <<>> twitter.com +trace any',
      ';; global options: +cmd',

It doesn't contain any useful info, just shows what tool we used and what request sent. I believe there's no sense to show it to users.

File: src/utils/index.ts (hunk 1 above)
Comment (poludnev) [reply src/utils/index.ts:148]: We can and we would, but I the near future I want to unify form elements. Now I had to make two: one form for tweet id, one for url (previous edition). I url form I used yup package for validation, and its validation schemas are async. 
Make a custom validation schema for tweet id as a bigint require some time, so I gave it up for now, and keep validator async.

File: src/components/TweetDetailsPreview/TweetDetailsPreview.tsx [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+
+const renderBodyElements = (key: keyof ITweetBody, body: ITweetBody) => {
+
+  const bodyCardKeys = body.card ? (Object.keys(body.card) as [keyof typeof body.card]) : [];
+  return (
+    <div className={classes.dataSubBlock} key={key + String(number)}>
+      <div className={classes.header}>{key}</div>
+      <div className={classes.value}>
+        {key === 'media' &&
+          (!!body[key] && body[key]!.length > 0 ? (
```
[the remark this answers is not in the document]
Comment (poludnev) [reply src/components/TweetDetailsPreview/TweetDetailsPreview.tsx]: fixed

**Pull request merged**